libopenage/util/hash.cpp

```diff
@@ -11,7 +11,7 @@ namespace util {
 
 size_t hash_combine(size_t hash1, size_t hash2) {
 	// Taken from http://www.boost.org/doc/libs/1_55_0/doc/html/hash/reference.html#boost.hash_combine
-	return hash1 ^ (hash2 + 0x9e3779b9 + ((hash1 << 6) + (hash1 >> 2)));
+	return ( hash1 + 0x9e3779b9 ) ^ ( hash2 + 0x9e3779b9 + ( ( hash1 + 0x9e3779b9 ) << 6 ) + ( ( hash1 + 0x9e3779b9 ) >> 2 ) );
```

Comment:
in the new version, the implementation for `size_t` is still the same, what you inserted here is the int32 one:
* https://www.boost.org/doc/libs/1_70_0/doc/html/hash/reference.html#boost.hash_combine
* https://www.boost.org/doc/libs/1_70_0/boost/container_hash/hash.hpp

The correct one is this (which is the one we had before)`:
``` cpp
        template <typename SizeT>
        inline void hash_combine_impl(SizeT& seed, SizeT value)
        {
            seed ^= value + 0x9e3779b9 + (seed<<6) + (seed>>2);
        }
```

Reply:
changed it back